src/psyclone/psyir/nodes/array_mixin.py

```diff
@@ -598,6 +598,121 @@ def get_outer_range_index(self):
                 return child.position
         raise IndexError
 
+    def same_range(self, index: int, array2, index2: int) -> bool:
+        ''' This method compares the range length of this array node
+        at a given index with the range length of a second array at
+        a second index. This is useful to verify if array operations
+        are valid, e.g.: A(3,:,5) + B(:,2,2).
+
+        Note that this check supports symbolic comparisons, e.g.:
+        A(3:4) has the same range as B(2+1:5-1),
+        and will consider compile-time unknown dimensions as equal, e.g.:
+        A(:) has the same range as B(:).
+
+        TODO #2485. This method has false negatives: cases when the range
+        is the same but it can not be proved, so we return False.
+
+        :param index: the index indicating the location of a range node in
+            this array.
+        :param array2: the array accessor that we want to compare it to.
+        :param index2: the index indicating the location of a range node in
+            array2.
+
+        :returns: True if the ranges are the same and False if they are not
+        the same, or if it is not possible to determine.
```

Comment:
The continuation line needs to be indented, otherwise the ref-guide complains.

Reply:
Done